packages/vanilla/src/sass/base/_core.scss

```diff
@@ -23,7 +20,7 @@ button {
 
 [aria-disabled=true],
 :disabled,
-input:read-only,
+input:read-only:not([type="checkbox"], [type="radio"], [type="file"], [type="range"], [type="color"]),
```

Comment:
is this necessary? Why would you not use the not-allowed cursor on these input types if they are disabled? Or are they showing up because of the "readonly" aspect?

Reply:
These are necessary because the not-allowed cursor is showing up on these form controls when they are non-disabled, these form controls do not have a `readonly` attribute and no `:readonly` pseudo-class in their css.